services/src/main/java/org/jboss/windup/web/services/RuleDataLoader.java

```diff
@@ -134,11 +134,27 @@ private void loadRules(Configuration webConfiguration)
                         continue;
                 }
                 LOG.info("Purging existing rule data for: " + rulesPath);
-                // Delete the previous ones
-                entityManager
-                            .createNamedQuery(RuleProviderEntity.DELETE_BY_RULES_PATH)
+
+                logCurrentEntityCounts("1");
+                {
+                    // Delete the previous providers from that path
+                    List<RuleProviderEntity> providers = entityManager.createNamedQuery(RuleProviderEntity.SELECT_BY_RULES_PATH, RuleProviderEntity.class)
```

Comment:
I had some comment in old PR to make a function from these 2 to avoid repeated code.